numpy/ma/tests/test_core.py

```diff
@@ -4857,11 +4857,16 @@ def test_immutable(self):
         assert_raises(ValueError, operator.setitem, view, (), 1)
         assert_raises(ValueError, operator.setitem, view.data, (), 1)
         assert_raises(ValueError, operator.setitem, view.mask, (), False)
-
-    @dec.knownfailureif(sys.version_info.major == 2, "See gh-9751")
+        
     def test_coercion_int(self):
         a_i = np.zeros((), int)
         assert_raises(MaskError, operator.setitem, a_i, (), np.ma.masked)
+    
+    @dec.knownfailureif(sys.version_info.major == 3, "long doesn't exist "
+                                                     "in Python 3")
+    def test_coercion_long(self):
+        a_l = np.zeros((), long)
+        assert_raises(MaskError, operator.setitem, a_l, (), np.ma.masked)
```

Comment:
This test is kinda unnecessary, as it actually tests exactly the same thing as above. `long(np.ma.masked)` would be a good test to have somewhere though.

Also, you want `skipif`, not `knownfailureif` here - the latter is reserved for things that make sense to fix in future.

Reply:
So if I test with an `int` in nose, it automatically tests `int` for Python 3 and `int` and `long` for Python 2? This means I should just get rid of my new testing function and I'm done? I was already wondering why you added a `knownfailureif` decorator for `test_coercion_int`, this explains it.

Reply:
`nose` isn't doing anything special here.

The old bug was not actually caused by `long` there, but by any integer type, even `np.int8` - because _internally_, numpy was calling `long`.

It would still be good to have a `long(np.ma.masked)` test somewhere. I'd hope we already have one for `int` which it could go alongside, but I might be overly optimistic there.